x-pack/plugins/security_solution/public/cases/components/create/form_context.test.tsx

```diff
@@ -0,0 +1,470 @@
+
+import { ConnectorTypes } from '../../../../../case/common/api';
+import { TestProviders } from '../../../common/mock';
+import { usePostCase } from '../../containers/use_post_case';
+import { useGetTags } from '../../containers/use_get_tags';
+import { useConnectors } from '../../containers/configure/use_connectors';
+import { useCaseConfigure } from '../../containers/configure/use_configure';
+import { connectorsMock } from '../../containers/configure/mock';
+import { useGetIncidentTypes } from '../settings/resilient/use_get_incident_types';
+import { useGetSeverity } from '../settings/resilient/use_get_severity';
+import { useGetIssueTypes } from '../settings/jira/use_get_issue_types';
+import { useGetFieldsByIssueType } from '../settings/jira/use_get_fields_by_issue_type';
+import { useCaseConfigureResponse } from '../configure_cases/__mock__';
+import {
+  sampleConnectorData,
+  sampleData,
+  sampleTags,
+  useGetIncidentTypesResponse,
+  useGetSeverityResponse,
+  useGetIssueTypesResponse,
+  useGetFieldsByIssueTypeResponse,
+} from './mock';
+import { FormContext } from './form_context';
+import { CreateCaseForm } from './form';
+import { SubmitCaseButton } from './submit_button';
+
+jest.mock('../../containers/use_post_case');
+jest.mock('../../containers/use_get_tags');
+jest.mock('../../containers/configure/use_connectors');
+jest.mock('../../containers/configure/use_configure');
+jest.mock('../settings/resilient/use_get_incident_types');
+jest.mock('../settings/resilient/use_get_severity');
+jest.mock('../settings/jira/use_get_issue_types');
+jest.mock('../settings/jira/use_get_fields_by_issue_type');
+jest.mock('../settings/jira/use_get_single_issue');
+jest.mock('../settings/jira/use_get_issues');
+
+const useConnectorsMock = useConnectors as jest.Mock;
+const useCaseConfigureMock = useCaseConfigure as jest.Mock;
+const usePostCaseMock = usePostCase as jest.Mock;
+const useGetIncidentTypesMock = useGetIncidentTypes as jest.Mock;
+const useGetSeverityMock = useGetSeverity as jest.Mock;
+const useGetIssueTypesMock = useGetIssueTypes as jest.Mock;
+const useGetFieldsByIssueTypeMock = useGetFieldsByIssueType as jest.Mock;
+const postCase = jest.fn();
+
+const defaultPostCase = {
+  isLoading: false,
+  isError: false,
+  caseData: null,
+  postCase,
+};
+
+const fillForm = async (wrapper: ReactWrapper) => {
+  await act(async () => {
+    wrapper
+      .find(`[data-test-subj="caseTitle"] input`)
+      .first()
+      .simulate('change', { target: { value: sampleData.title } });
+  });
+
+  await act(async () => {
+    wrapper
+      .find(`[data-test-subj="caseDescription"] textarea`)
+      .first()
+      .simulate('change', { target: { value: sampleData.description } });
+  });
+
+  await waitFor(() => {
+    ((wrapper.find(EuiComboBox).props() as unknown) as {
+      onChange: (a: EuiComboBoxOptionOption[]) => void;
+    }).onChange(sampleTags.map((tag) => ({ label: tag })));
+  });
+};
```

Comment:
this does not need to be async, and the waitFor should be an act. you do not use `await` with `act`. you'll also need to remove the `await` in front of the calls to `fillForm`

```suggestion
const fillForm = (wrapper: ReactWrapper) => {
  wrapper
    .find(`[data-test-subj="caseTitle"] input`)
    .first()
    .simulate('change', { target: { value: sampleData.title } });
  wrapper
    .find(`[data-test-subj="caseDescription"] textarea`)
    .first()
    .simulate('change', { target: { value: sampleData.description } });

  act(() => {
    ((wrapper.find(EuiComboBox).props() as unknown) as {
      onChange: (a: EuiComboBoxOptionOption[]) => void;
    }).onChange(sampleTags.map((tag) => ({ label: tag })));
  });
};
```

Reply:
`await` with `act` is not invalid. It is supported since `React 16.9`. But you are right. It is not needed for this scenario.